### PR TITLE
Add cryptography

### PR DESCRIPTION
**Pull request by jschueller**
Updated from #1333, have to run pytest after setup as workdir is deleted :[
needs https://github.com/conda-forge/cryptography-vectors-feedstock/pull/7

### CODE REVIEW

**Comment by jakirkham** [comment]
There's one failure on AppVeyor, but it seems it can be ignored. Would disable it or patch it somehow.

Weird the others don't encounter it though. 😕

xref: https://github.com/pyca/cryptography/issues/3196

**Comment by jschueller** [comment]
Maybe it will go well with openssl > 102i now that 102k is published.

**Comment by jschueller** [comment]
Indeed, the openssl update made it work!

**Comment by jschueller** [comment]
I think it's ready now.

File: recipes/cryptography/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    - toolchain
+    - python
+    - setuptools >=11.3
+    - openssl 1.0.*
+    - cffi >=1.7
+    - enum34  # [py2k]
+    - ipaddress  # [py2k]
+    - asn1crypto >=0.21.0
+    - six >=1.4.1
+    - idna >=2.1
```
Comment (jakirkham) [inline recipes/cryptography/meta.yaml]: Do we need all of these at build time or can some of them be dropped?

File: recipes/cryptography/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    - pytz
+  source_files:
+    - tests
+  commands:
+    - pytest
+
+about:
+  home: https://github.com/pyca/cryptography
+  license: Apache 2.0 or BSD 3-Clause
+  #license_file: ??? TWO LICENSE FILES ???
```
Comment (jakirkham) [inline recipes/cryptography/meta.yaml]: Appears they added a third one. For now I'd just use `LICENSE`, which summarizes all of the licenses in play.

**Comment by jakirkham** [comment]
Couple minor comments. Otherwise LGTM.

File: recipes/cryptography/meta.yaml (hunk 2 above)
Comment (jschueller) [inline recipes/cryptography/meta.yaml]: ok

File: recipes/cryptography/meta.yaml (hunk 1 above)
Comment (jschueller) [inline recipes/cryptography/meta.yaml]: seems I can drop most except cffi & openssl

Comment (jakirkham) [inline recipes/cryptography/meta.yaml]: Yep that seems reasonable.

Side note: GitHub has moved this comment out of context. It was in the `build` section, but the issue raised has been subsequently addressed.

File: recipes/cryptography/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    - pytest >=2.9.0
+    - pytz
+  source_files:
+    - tests
+  commands:
+    - pytest
+
+about:
+  home: https://github.com/pyca/cryptography
+  license: Apache 2.0 or BSD 3-Clause
```
Comment (jakirkham) [inline recipes/cryptography/meta.yaml]: Forgot to mention. Maybe it is worth mentioning the PSF section of code? Then again, this is noted in the license.

Comment (jschueller) [inline recipes/cryptography/meta.yaml]: ok, added

File: recipes/cryptography/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    - pytest >=2.9.0
+    - pytz
+  source_files:
+    - tests
+  commands:
+    - pytest
+
+about:
+  home: https://github.com/pyca/cryptography
+  license: Apache 2.0 or BSD 3-Clause or PSF
```
Comment (jakirkham) [inline recipes/cryptography/meta.yaml]: Ah sorry, I said that in a confusing way. So the codebase is still Apache 2.0 or BSD 3-Clause, but the PSF portion is always PSF. So maybe drop `or` and use a `,` to indicate it is just one of the licenses. Also, as for the license version, this is PSF 2.

TL;DR Please change to `Apache 2.0 or BSD 3-Clause, PSF 2`

Comment (jschueller) [inline recipes/cryptography/meta.yaml]: ok

**Comment by jakirkham** [comment]
Thanks @jschueller.
